src/Resolver/File.php

```diff
@@ -81,6 +81,16 @@ public function resolve($definition)
             $path = realpath($this->getIterator()->getRootDefinition()->getBasepath() . \DIRECTORY_SEPARATOR . $path);
         }
 
-        return file_get_contents($path);
+        $content = file_get_contents($path);
+
+        if (pathinfo($path, PATHINFO_EXTENSION) == 'json') {
```

Comment:
We can't assume we need to parse based on file extension, should have this logic:

1. `$parse === 'auto' && pathinfo($path, PATHINFO_EXTENSION) == 'json'` (auto parse)
2. `$parse === 'json'` (explicit parse)

I think the first gate covers it, but `$parse === 'text'` should return raw content, to allow raw json to still be returned if needed.

Reply:
Whoops! Good catch